db/db_iter.cc

```diff
@@ -709,6 +732,12 @@ bool DBIter::FindValueForCurrentKey() {
   size_t num_skipped = 0;
   while (iter_->Valid() && ikey.sequence <= sequence_ &&
          user_comparator_->Equal(ikey.user_key, saved_key_.GetKey())) {
+    if (TooManyInternalKeysSkipped()) {
+      return false;
+    } else {
+      num_internal_keys_skipped_++;
+    }
+
```

Comment:
@lightmark should there be something different done for merges? I am assuming that they should also be handled in the same way. We should still fail and return an incomplete status if too many keys are being looked for getting the value of a merge.  Thoughts?